src/include/daos/checksum.h

```diff
@@ -465,15 +469,12 @@ csum_chunk_count(uint32_t chunk_size, uint64_t lo_idx, uint64_t hi_idx,
 		 uint64_t rec_size);
 
 static inline bool
-csum_iod_is_supported(uint64_t chunksize, daos_iod_t *iod)
+csum_iod_is_supported(daos_iod_t *iod)
 {
 	/**
-	 * iod_size must be greater than 1 and chunksize must be larger
-	 * than iod size if it's an array type. Doesn't support very large
-	 * record size yet for array types
+	 * iod_size must be greater than 1
 	 */
-	return iod->iod_size > 0 &&
-	       (iod->iod_type == DAOS_IOD_SINGLE || iod->iod_size <= chunksize);
+	return iod->iod_size > 0;
```

Comment:
So there is no checksum for punched record, right? then how to detect bad non-punch record that become punched record if iod_size is corrupted?

Reply:
Good question. Right now there is no detection of corruption of any of the iod. Something we might need to consider, but outside the scope of this change.